### PR TITLE
Keycloak Onboarding Documentation

### PR DESCRIPTION
**Pull request by akroon3r**
# Description

This PR includes the following proposed change(s):

[FLA-365](https://justice.gov.bc.ca/jira/browse/FLA-365)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Not tested.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/onboarding.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Getting Started - Onboarding
+
+## Request Keycloak Client
+
+- create github issue (link to issue) for keycloak client creation (TBD)
+- on completetion of github issue, the user will be provided a *client_id*, *realm*, *client_secret*. All of which are required to be kept secured at all times.
```
Comment (sdevalapurkar-bcgov) [inline docs/onboarding.md:6]:
```suggestion
- on completion of github issue, the user will be provided a *client_id*, *realm*, *client_secret*. All of which are required to be kept secured at all times.
```

File: docs/onboarding.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Request Keycloak Token
+
+First, users will want to POST against the following URL: ${keycloakBaseUrl}/realms/${keycloakRealm}/protocol/openid-connect/token
+
+- ${keycloakBaseUrl} will be the value https://sso-${ENVIRONEMENT}.pathfinder.gov.bc.ca/auth , where ${ENVIRONMENT} is either dev, test or prod
```
Comment (sdevalapurkar-bcgov) [inline docs/onboarding.md:12]:
```suggestion
- ${keycloakBaseUrl} will be the value https://sso-${ENVIRONEMENT}.pathfinder.gov.bc.ca/auth, where ${ENVIRONMENT} is either dev, test or prod
```

File: docs/onboarding.md (hunk 1 above)
Comment (alexjoybc) [inline docs/onboarding.md]:
```suggestion
On completetion of github issue, the user will be provided [OpenId Connect](https://openid.net/connect/) information such  as *client_id*, *realm*, *client_secret*. All of which are required to be kept secured at all times.
```

**Comment by akroon3r** [comment]
@sdevalapurkar-bcgov tyty

File: docs/onboarding.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Getting Started - Onboarding
+
+## Request Keycloak Client
+
+- create github issue (link to issue) for keycloak client creation (TBD)
```
Comment (alexjoybc) [inline docs/onboarding.md:5]:
```suggestion
Create github issue (link to issue) for keycloak client creation (TBD)
```

File: docs/onboarding.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+In the request body, the following information will be required
+
+Example body: client_id=${keycloakClientId}&grant_type=client_credentials&client_secret=${keycloakClientSecret}
+
+- client_id : Provided to the user by the admin after the keycloak client request has been fufilled
+- grant_type : value will always be `client_credentials`
+- client_secret : Provided to the user by the admin after the keycloak client request has been fufilled
+
+Once the request for the keycloak token has been completed successfully, the user can begin to interface with the Efiling Hub [API](https://editor.swagger.io/?url=https://raw.githubusercontent.com/bcgov/jag-file-submission/master/src/backend/efiling-api/jag-efiling-api.yaml) and Frontend Applications. The two API endpoints that a parent application will interface with are the `/submission/documents` and `/submission/${submissionId}/generateUrl`.
```
Comment (alexjoybc) [inline docs/onboarding.md:23]: could you add the following curl:

```
curl --location --request POST 'https://[keycloak auth endpoint]/auth/realms/[keycloak-realm]/protocol/openid-connect/token' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'client_id=[your client id]' \
--data-urlencode 'grant_type=client_credentials' \
--data-urlencode 'client_secret=[your client secret]'
```

File: docs/onboarding.md (hunk 1 above)
Comment (akroon3r) [reply docs/onboarding.md]: @alexjoybc can you re suggest and I'll commit.

File: docs/onboarding.md (hunk 4 above)
Comment (akroon3r) [reply docs/onboarding.md:23]: done

File: docs/onboarding.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# Getting Started - Onboarding
+
+## Request Keycloak Client
+
+Create Github [issue]() to begin the onboarding process to our application.
```
Comment (sdevalapurkar-bcgov) [inline docs/onboarding.md:5]:
```suggestion
Create a Github [issue]() to begin the onboarding process to our application.
```

File: docs/onboarding.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+On completetion of github issue, the user will be provided [OpenId Connect](https://openid.net/connect/) information such  as *client_id*, *realm*, *client_secret*. All of which are required to be kept secured at all times.
```
Comment (sdevalapurkar-bcgov) [inline docs/onboarding.md:7]:
```suggestion
On completion of the Github issue, the user will be provided [OpenId Connect](https://openid.net/connect/) information such  as *client_id*, *realm*, *client_secret*. All of which are required to be kept secured at all times.
```

**Pull request merged**